File: src/main/java/me/pljr/pljrapispigot/config/SoundType.java
package me.pljr.pljrapispigot.config;

import com.cryptomorin.xseries.XSound;
import me.pljr.pljrapispigot.builders.SoundBuilder;
import me.pljr.pljrapispigot.managers.ConfigManager;
import me.pljr.pljrapispigot.objects.PLJRSound;
import org.bukkit.configuration.file.FileConfiguration;

import java.util.HashMap;

public enum SoundType {
    COMMAND_FAIL(new SoundBuilder(XSound.ENTITY_VILLAGER_NO, 10, 1).create()),
    TELEPORT_TICK(new SoundBuilder(XSound.UI_BUTTON_CLICK, 10, 1).create()),
    TELEPORT_FAIL(new SoundBuilder(XSound.ENTITY_VILLAGER_HURT, 10, 1).create()),
    TELEPORT_TP(new SoundBuilder(XSound.ENTITY_CAT_PURR, 10, 1).create());

    private static HashMap<SoundType, PLJRSound> soundType;
    private final PLJRSound defaultValue;

    SoundType(PLJRSound defaultValue){
        this.defaultValue = defaultValue;
    }

    public static void load(ConfigManager config){
        FileConfiguration fileConfig = config.getConfig();
        soundType = new HashMap<>();
        for (SoundType soundType : values()){
            if (!fileConfig.isSet(soundType.toString())){
                config.setPLJRSound(soundType.toString(), soundType.defaultValue);
            }else{
                SoundType.soundType.put(soundType, config.getPLJRSound(soundType.toString()));
            }
        }
        config.save();
    }

    public PLJRSound get(){
        return soundType.getOrDefault(this, defaultValue);
    }
}

File: src/main/kotlin/me/pljr/pljrapispigot/config/configuration/SoundType.kt
package me.pljr.pljrapispigot.config.configuration

import com.cryptomorin.xseries.XSound
import me.pljr.pljrapispigot.config.ConfigManager
import me.pljr.pljrapispigot.sound.PLJRSound
import me.pljr.pljrapispigot.sound.SoundBuilder
import org.bukkit.configuration.file.FileConfiguration
import java.util.*

enum class SoundType(val defaultValue: PLJRSound) {
    COMMAND_FAIL(SoundBuilder(XSound.ENTITY_VILLAGER_NO, 10f, 1f).create()),
    TELEPORT_TICK(SoundBuilder(XSound.UI_BUTTON_CLICK, 10f, 1f).create()),
    TELEPORT_FAIL(SoundBuilder(XSound.ENTITY_VILLAGER_HURT, 10f, 1f).create()),
    TELEPORT_TP(SoundBuilder(XSound.ENTITY_CAT_PURR, 10f, 1f).create());

    companion object {
        private val soundType = EnumMap<SoundType, PLJRSound>(SoundType::class.java)

        fun load(config: ConfigManager) {
            val fileConfig: FileConfiguration = config.config
            soundType.clear()
            values().forEach {
                if (!fileConfig.isSet(it.toString())){
                    config.setPLJRSound(it.toString(), it.defaultValue)
                } else {
                    soundType[it] = config.getPLJRSound(it.toString())
                }
            }
            config.save()
        }
    }

    fun get(): PLJRSound {
        return soundType.getOrDefault(this, this.defaultValue)
    }
}